src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -126,6 +126,8 @@ void Run (DirectoryAssemblyResolver res)
 					Log.LogDebugMessage ($"Skipping Java Stub Generation for {asm.ItemSpec}");
 					continue;
 				}
+				if (!assemblies.All (x => Path.GetFileName (x) != Path.GetFileName (asm.ItemSpec)))
+					continue;
```

Comment:
Is this for removing duplicates? What changed that we need to do this now? Also, is this safe - with casing?

Reply:
This is a safety net. I had a problem with the ItemGroups passing in duplicate file entires. I fixed that, but we really should not be allowing duplicates here as it breaks the generator. This is just a "just in case" Its the same code as the foreach loop at bit further down.